### ISSUE TITLE
iPhone X full screen not supported

### ISSUE THREAD

**Issue by odbuser2** [javafxports/javafxmobile-plugin#39]
org.javafxports:jfxmobile-plugin:2.0.16

Black(blank) bars are present on the top and bottom in portrait mode and left and right in landscape mode on the iPhone X.  Can support be added for the latest devices?  Is this simply a gvm recompile against the latest XCode libraries?

**Comment by odbuser2** [comment]
I'm sure the devs already know this but after March 31, iPhone X support is mandated:

https://developer.apple.com/ios/submit/
"Update your version of Xcode to the latest release of Xcode 9 available on the Mac App Store, which includes the iOS 11 SDK, and build your apps. Starting April 2018, all new iOS apps submitted to the App Store must be built with the iOS 11 SDK. All new apps for iPhone, including universal apps, must support the Super Retina display of iPhone X."

**Comment by jperedadnr** [comment]
Can you add a black image file named `Default-375w-812h@3x~iphone.png` with resolution 1125x2436 to you src/ios/assets and try again?

**Comment by odbuser2** [comment]
Yes, that worked!  I see that the Gluon Netbeans plugin (2.6.2) is missing this asset.

Does glisten account for the notch?  For glisten and non-glisten based apps, is there a way to know the exact orientation?  OrientationService only knows HORIZONTAL and VERTICAL but the notch is only on one side...

**Comment by jperedadnr** [comment]
The plugins need to be updated with the missing resource, indeed.

For now, Glisten is not aware of the notch. The OrientationService already deals internally with left, right [orientation](https://bitbucket.org/gluon-oss/charm-down/src/0ac7ae0a8ed83ecbee362b8b4adc91574aa56785/plugins/plugin-orientation/ios/src/main/java/com/gluonhq/charm/down/plugins/ios/IOSOrientationService.java?at=default&fileviewer=file-view-default#IOSOrientationService.java-69), but since there is no equivalent on Android, it doesn't expose those.

**Comment by odbuser2** [comment]
...but aren't notches coming in Android?  I suggest to use the following in OrientationService instead of javafx.geometry.Orientation directly:
```
import javafx.geometry.Orientation;

public enum DeviceOrientation {
    HORIZONTAL_LEFT(Orientation.HORIZONTAL),
    HORIZONTAL_RIGHT(Orientation.HORIZONTAL),
    VERTICAL_UP(Orientation.VERTICAL),
    VERTICAL_DOWN(Orientation.VERTICAL);
    
    Orientation orientation;
    DeviceOrientation(Orientation orientation) {
        this.orientation = orientation;
    }
    public Orientation getOrientation() {
        return this.orientation;
    }
}
```

**Comment by odbuser2** [comment]
Turns out this is really needed... on iOS 11 on an iPhone X, turning the phone upside down triggers an Orientation event but does not rotate the scene.  This will likely result in the developers coding to pad the top on the event but since it does not rotate, it stay in landscape with extra padding on the or right of the display (which is actually the top of the scene).

I checked a bunch of native apps and it seems that the general behavior is to leave it in the previous orientation, thus ignoring the orientation when the device is "upside down".
